rclcpp/include/rclcpp/clock.hpp

```diff
@@ -88,6 +88,16 @@ class Clock
    *     false. There is not a consistent choice of sleeping time when the time source changes,
    *     so this is up to the caller to call again if needed.
    *
+   * \warning When using gcc < 10 or gcc >= 10 and pthreads lacks the function
```

Comment:
Reading fast this I was confused about which versions of gcc are affected since with gcc < 10 | gcc >= 10 my mind understand all versions .

Reply:
In 470f31f2 I tried to separate the gcc versions by adding another `when using` to the sentence. Does that help some?

Reply:
much better thanks.